### ISSUE TITLE
unable to adjust width to 100% using custom.css for terminal.

### ISSUE THREAD

**Issue by cailiang9** [jupyter/notebook#46]
sometimes it's 100%, and then shrink to 40 chars width the next time open it (It's 100% black but only display chars in left 40 chars).

**Comment by Carreau** [comment]
I wouldn't recommend trying to resize the terminal using css, the sizing is a complex topic.

**Comment by cailiang9** [comment]
Thanks! I'm not trying to resize it dynamically. Is there a way to adjust ?

**Comment by Carreau** [comment]
you will have to play around the follwing in main.js:
```
    window.onresize = function() { 
      var geom = calculate_size();
      terminal.term.resize(geom.cols, geom.rows);
      terminal.socket.send(JSON.stringify(["set_size", geom.rows, geom.cols,
                                    $(window).height(), $(window).width()]));
    };
```
terminal il a local variable, so might be difficult to get access to.

**Comment by takluyver** [comment]
We should probably expose `terminal` globally so people can fiddle with it, as we do for `IPython`.

**Comment by takluyver** [comment]
See PR #48

**Comment by JamiesHQ** [comment]
@takluyver : is this issue closed by PR #48?

**Comment by takluyver** [comment]
I suspect it still doesn't work with just an extra CSS rule, as @cailiang9 was asking for. But given the complexities of the terminal, I don't think it can be resized without fiddling with Javascript. So I think this is as fixed as it's going to get.
